solver.md

```diff
@@ -0,0 +1,461 @@
+---
+layout: index
+title: GCAM Solver
+prev: 
+next:
+current-version: v4.2 
+---
+
+At each time step, GCAM searches for a vector of prices that casue all
+markets to be cleared and all consistency conditions to be satisfied.
+The mapping from input prices to output market disequilibria is a
+vector function $$\vec y = F(\vec p)$$.  The GCAM solver is responsible
+for finding the root of this equation; that is, the point at which
+$$F(\vec p) = 0$$.
+
+GCAM has several solver algorithms at its disposal. The solver
+algorithms can be combined so that several of them are used in
+sequence. The mix of algorithms can be varied from one model timestep
+to the next and can be customized for markets that require special
+treatment. Additionally, each solver algorithm has several adjustable
+parameters that are user configurable. These configuration options are
+specified in the
+[solver configuration file](#Solver-Configuration-File).
+
+Solver Algorithms
+-----------------
+
+### Bisection Solver
+
+#### Description
+
+The bisection solver is a multidimensional generalization of of
+one-dimensional bisection methods. The one-dimensional methods work by
+first establishing a bracket, identified by a change of sign in
+$$f(x)$$, around the solution. At each iteration $$f(x)$$ is evaluated
+at the midpoint of the bracket interval, and the sign of the result
+tells us which half of the interval contains the solution. The
+endpoints of the bracket are adjusted to the newly-identified
+interval, and the iterations continue until the interval is small
+enough that we have effectively isolated the solution.
+
+For a system of equations in multiple dimensions, it is generally not
+possible to construct a rigorous bracket around a solution. Even if we
+find two points for which all of the equations in the system have
+opposite sign, there is no guarantee that a solution exists somewhere
+on the line connecting the two points. The reason why is that the
+solution to the equation $$\vec F(\vec x) = 0$$ is a single point
+$$\vec x_0$$ where all components of $$\vec F$$ are simultaneously
+zero. As we move from one of our putative bracket points to the other
+the overwhelming likelihood is that the components of $$\vec F$$ will
+change sign at different places along the path, none of which will be
+the solution we seek. Thus, in general, bracketing and bisection is
+not a viable strategy for solving a system of equations.
+
+Despite this shortcoming, bisection can often be used to get the
+solver into the general vicinity of the solution, from which one of the
+other solvers can easily find it. For initial guesses far from
+the solution, bisection can often be faster than more sophisticated
+solution techniques. Moreover, bisection places no conditions on the
+Jacobian matrix, making it useful for jostling the solver out of
```

Comment:
These paragraphs are interesting and well-written. Here, perhaps briefly define (not so much mathematically, as its role) "Jacobian matrix" for the reader.